tools/postgen.ts

```diff
@@ -0,0 +1,59 @@
+import * as fs from 'fs';
+const recursive = require("recursive-readdir");
+
+/**
+ * This tool makes the generated code from the quicktype wrapper usable to TypeScript clients.
+ * 
+ * Prerequisites:
+ * - A `cloud/` folder with `*Data.ts` files within subfolders.
+ * 
+ * TypeScript "toTYPE" function:
+ * - Loop through every type (ts file)
+ *   - 1st interface: wrapper type (ignore)
+ *   - 2nd interface: main type (use this). i.e. AuditLogWrittenEvent
+ * - Add a to[TYPE] function to the file. Example:
+ *     export const toMessagePublishedEvent = (json: object) => {
+ *       return json as MessagePublishedEvent;
+ *     };
+ */
+
+// Wrap in IIFE for top-level await
+(async () => {
+  const filePaths: string[] = await recursive("cloud");
+
+  // Get every data type file path
+  const dataTsFilePaths = filePaths.filter((path) => {
+    const typeFilenamePostfix = 'Data.ts'
+    return path.endsWith(typeFilenamePostfix);
+  });
+  
+  // Read type file
+  dataTsFilePaths.map(filename => {
+    // For a single type file
+    console.log(`- Fixing: ${filename}`);
+    const typeFileContent = fs.readFileSync(filename).toString();
+
+    // Get 2nd interface
+    const lines = typeFileContent.split('\n');
+    const interfaceLines = lines.filter((line) => {
+      return line.startsWith('export interface');
+    });
+
+    // The data name, e.g. 'export interface AuditLogWrittenEvent {'
+    const dataEventName = interfaceLines[1]
+        .split(' ')
+        .filter((token) => token.endsWith('Event'))[0];
+
+    const newTypeFileContent = typeFileContent + `
+/**
+ * Cast a raw JSON object to a typed event (useful for IDE autocompletion).
+ * @param {object} json The JSON object
+ * @return {${dataEventName}} The object with type annotations
```

Comment:
nit: Should this include a comment suggesting PRs to change this function should target this file?

Reply:
Not sure what we mean here. I hope things are not too complicated to change things?

Reply:
Once this file is generated, it won't be obvious to contributors that PRs to this function will be rejected in favor of changes to the postgen script. Noting in-line that this is generated will help that by prompting the developer to look elsewhere. Something like:

```
This function is generated, for details read https://github.com/googleapis/google-cloudevents-nodejs/tree/master/contributing.md#typed-event-function
```

As said, this is a nit and can easily be a follow-up/left pending on proof it's a contributor management overhead problem.